Sources/StreamChat/Controllers/DataController.swift

```diff
@@ -28,6 +28,15 @@ public class DataController: Controller {
             }
         }
     }
+
+    var isAvailableOnRemote: Bool {
+        switch state {
+        case .remoteDataFetched, .remoteDataFetchFailed:
+            return true
+        case .initialized, .localDataFetched, .localDataFetchFailed:
+            return false
+        }
+    }
```

Comment:
> Are we using this anywhere right now?
Yes, we are. In the creation of the Operations to reset queries and watch channels. Inside SyncRepository's `syncLocalState`

> "isAvailableOnRemote" but if it failed the remote call it will be true
This property is actually about whether this element is available in remote, as per the backend already having it. In the case of `.remoteDataFetchFailed` we can assume backend knows about it, but we failed to fetch it.